### DISCUSSION TITLE
Restoring Seq from backup

### DISCUSSION THREAD

**Discussion by agermalid** [datalust/seq-tickets#1694] [Help] [answered]
Hi,

I can't restore a Seq backup from a test docker container to a new one. I did everything according to official article: [Backup and Restore](https://docs.datalust.co/docs/backup-and-restore-docker) and I can't figure out what the problem is.

I have a backup file: seq_20221121.seqbac

docker-compose.yaml for the new Seq:

> version: "3.7"
services:
  seq:
    image: datalust/seq:latest
    container_name: seq-local
    environment:
      ACCEPT_EULA: "Y"
    volumes:
      - "seq_data:/data"
      ~"./seq_20221121.seqbac:/data/Backups/seq_20221121.seqbac"~
       ~"./restore_script.sh:/data/Init/restore_script.sh"~
    ports:
      - "5342:80"
      - "5341:5341"
volumes:
     seq_data:

restore_script.sh:
`seqsvr restore --secret-key="8D2RNAbAgmV7ur8nQCLT7PlCWCk+kW488nSg3Narj/4=" --backup="/data/Backups/seq_20221121.seqbac" --enable-apps`

In Seq docker container nothing happens:
> ────────────────────────────────────────
  Seq ♦ Machine data, for humans.
─────────── © 2022 Datalust Pty Ltd ────
Running as server; press Ctrl+C to exit.
[10:20:21 INF] Seq "2022.1.7929" running on OS "Linux 5.10.0-19-amd64 #1 SMP Debian 5.10.149-2 (2022-10-21)"
[10:20:21 INF] Seq detected 2010.804224 MB of RAM
[10:20:22 INF] Opening event store at "/data/Stream/stream.flare"
[10:20:22 INF] Ingestion enabled
[10:20:22 INF] Opening metastore "/data/Documents/metastore.flare"
[10:20:22 INF] Storage subsystem available
[10:20:22 INF] Seq listening on ["http://localhost/", "https://localhost/", "http://localhost:5341/", "https://localhost:45341/"]

~In /data/Logs/seq-20221122.log no erros or words like: "restore"~
~[seq-20221122.log](https://github.com/datalust/seq-tickets/files/10066546/seq-20221122.log)~

**IMPORTANE! - you need to copy restore_script.sh and seq_20221121.seqbac only after Seq container was started, then restart the container and after that create Backup directory at /data/:**

`docker compose up -d`
`docker exec -it seq-local mkdir /data/Backups`
`docker cp seq_20221121.seqbac seq-local:/data/Backups/`
`docker cp restore_script.sh seq-local:/data/Init/`
`docker compose restart seq-local`

**Comment by nblumhardt** [comment, marked as answer]
Hi! Your `restore` command line and configuration look good to me 👍 

If you've previously run an init script with the same name, then the subsequent one won't be executed - you'd need to give your new restore script a unique name.

My bet though is that your script and backup aren't being mounted as files in the container - Docker volume mounts are generally directories, I think individual files may work if they already exist (trawling other Q&A out there) but it's a bit of a long shot.

Exec-ing into the container and `ls`-ing the `/data/Init` directory would be a good idea to confirm what is being mounted, but putting those files in their respective locations in your `seq_data` directory would be the sure-fire way to go.

Hope this helps!
